src/runtime/methodbinder.cs

```diff
@@ -275,19 +282,42 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
 
                     for (int n = 0; n < clrnargs; n++) {
                         IntPtr op;
-                        if (n < pynargs)
-                        {
-                            if (arrayStart == n)
-                            {
+                        if (n < pynargs) {
+                            if (arrayStart == n) {
                                 // map remaining Python arguments to a tuple since
                                 // the managed function accepts it - hopefully :]
                                 op = Runtime.PyTuple_GetSlice(args, arrayStart, pynargs);
                             }
-                            else
-                            {
+                            else {
                                 op = Runtime.PyTuple_GetItem(args, n);
                             }
-                            Type type = pi[n].ParameterType;
+                            type = null;
+                            if  (_methods.Length>1) { 
+                                IntPtr pyoptype = IntPtr.Zero;
+                                //try {
+                                    pyoptype = Runtime.PyObject_Type(op);
+                                /*}
+                                catch (System.Exception) {
```

Comment:
deleted

On Tue, Feb 9, 2016 at 1:10 PM, Tony Roberts <notifications@github.com>
wrote:

> In src/runtime/methodbinder.cs
> <https://github.com/pythonnet/pythonnet/pull/137#discussion_r52357482>:
>
> >                                  op = Runtime.PyTuple_GetItem(args, n);
> >                              }
> > -                            Type type = pi[n].ParameterType;
> > +                            type = null;
> > +                            if  (_methods.Length>1) {
> > +                                IntPtr pyoptype = IntPtr.Zero;
> > +                                //try {
> > +                                    pyoptype = Runtime.PyObject_Type(op);
> > +                                /*}
> > +                                catch (System.Exception) {
>
> Please don't leave commented code - you commented it out because it was
> incorrect; it should be deleted instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pythonnet/pythonnet/pull/137/files#r52357482>.
>